### PR TITLE
fix(android): navigationWindow open/close event

### PR DESCRIPTION
**Pull request by m1ga**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-27834

##### TEST CASE
```JS
const win = Ti.UI.createWindow({ backgroundColor: 'white' });
const nav = Ti.UI.createNavigationWindow({ window: win });

nav.addEventListener('open', _ => {
	console.log('navigationWindow: open');
	nav.close()
});

nav.addEventListener('close', _ => {
	console.log('navigationWindow: close');
});

nav.open();
```
- Application will close
- Should see both navigation events in log

### CODE REVIEW

File: android/modules/ui/src/java/ti/modules/titanium/ui/NavigationWindowProxy.java [hunk 1]
```diff
@@ -32,6 +32,7 @@ public void open(@Kroll.argument(optional = true) Object arg)
 		// FIXME: Shouldn't this complain/blow up if window isn't specified?
 		if (getProperties().containsKeyAndNotNull(TiC.PROPERTY_WINDOW)) {
 			opened = true;
+			fireEvent(TiC.EVENT_OPEN, null);
```
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/NavigationWindowProxy.java:35]: These events will fire if the window is already open/closed. It's best to move the `fireEvent` into the `openWindow` and `closeWindow` methods.

**Review by garymathews — CHANGES_REQUESTED** [review]

File: android/modules/ui/src/java/ti/modules/titanium/ui/NavigationWindowProxy.java (hunk 1 above)
Comment (m1ga) [reply android/modules/ui/src/java/ti/modules/titanium/ui/NavigationWindowProxy.java:35]: but wouldn't that also fire when you run `$.navWindow.openWindow(child)`? Aren't those already captured in the childs window `open` event?

**Review by garymathews — APPROVED** [review]
CR: **PASS**

**Review by ssjsamir — APPROVED** [review]
FR Passed using the test case in the description above.

**Test Environment**
```
MacOS Catalina: 10.15.5 Beta
Xcode: 11.4
Java Version: 1.8.0_131
Android NDK: 21.1.6273396-beta2
Node.js: 10.16.3
""NPM":"5.0.0-1","CLI":"8.0.0-master.10""
Andoird Pixl xl 7.1.1 Emulator 
```

**Comment by sgtcoolguy** [comment]
Manually merged to master (had to fix up eslint issues, squash commits down)

See commits:
- ec1976a66127b6de024bb4fe6d20207e46fdcb4c
- 31b4d355c7b853517aea3564513f9866d9c29633

**Pull request closed without merging**